### PR TITLE
Add lowering of "aten.pow.Tensor_Scalar" op

### PR DESCRIPTION
**Pull request by Yutyrannus**

### CODE REVIEW

**Comment by pashu123** [comment]
Hey, @Yutyrannus Can you please add commit-description? `git commit -m "Title" -m "Description ..........";`

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp [hunk 1]
```diff
@@ -1375,6 +1375,17 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
...
   }
+  if (auto pow = dyn_cast<AtenPowTensorScalarOp>(op)) {
+    if (!pow.getType()
+             .cast<ValueTensorType>()
+             .getDtype()
+             .isa<mlir::FloatType>()) {
+      pow.emitError("unimplemented: non-floating point dtype");
+      return nullptr;
+    }
+    Value expF32 = b.create<arith::TruncFOp>(loc, b.getF32Type(), operands[1]);
```
Comment (pashu123) [inline lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1386]: Can you explain the reason for truncation?

Comment (Yutyrannus) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1386]: Torch.Float type is f64, math.pow need src and exponent date type is same, so cast exponent f64 -> f32.

**Comment by Yutyrannus** [comment]
commit-description has added

File: e2e_testing/torchscript/elementwise.py [hunk 2]
```diff
@@ -394,3 +394,19 @@ def forward(self, a):
...
+        None,
+        ([-1, -1], torch.float32, True),
+    ])
+
+    def forward(self, a):
+        return torch.pow(a, 2.0)
+
+@register_test_case(module_factory=lambda: ElementwisePowModule())
+def ElementwisePowModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4))
```
Comment (ramiro050) [inline e2e_testing/torchscript/elementwise.py:412]: nit: GitHub does not like it when files don't end in a newline

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 1 above)
Comment (ramiro050) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp:1386]: There's no guarantee that the input tensor is f32. I would instead use the helper

https://github.com/llvm/torch-mlir/blob/92ed64626b8e45f62ee0ae295b58f853a513b302/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1255

and do something like this to get the `dtype`

https://github.com/llvm/torch-mlir/blob/92ed64626b8e45f62ee0ae295b58f853a513b302/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L1467

**Review by ramiro050 — CHANGES_REQUESTED** [review]
Thanks for this!

A few comments below.

File: e2e_testing/torchscript/elementwise.py (hunk 2 above)
Comment (Yutyrannus) [reply e2e_testing/torchscript/elementwise.py]: THANKS! Code has updated

File: lib/Conversion/TorchToLinalg/TorchToLinalg.cpp (hunk 1 above)
Comment (Yutyrannus) [reply lib/Conversion/TorchToLinalg/TorchToLinalg.cpp]: THANKS! Code has updated

**Review by silvasean — APPROVED** [review]
Thanks for this! Thank you for contributing to Torch-MLIR! Let us know if you have any questions :)

**Pull request merged**